app/user.go

```diff
@@ -2258,15 +2275,32 @@ func (a *App) DemoteUserToGuest(user *model.User) *model.AppError {
 
 	demotedUser, err := a.GetUser(user.Id)
 	if err != nil {
-		return err
+		mlog.Error(err.Error())
 	}
 
-	message := model.NewWebSocketEvent(model.WEBSOCKET_EVENT_USER_UPDATED, "", "", "", nil)
-	message.Add("user", demotedUser)
-	a.Publish(message)
-
+	a.sendUpdatedUserEvent(*demotedUser)
```

Comment:
And here.

Reply:
Same case but yes, i prefer to be more defensive here. Thanks for the heads up!

Reply:
Done